src/sqlfluff/rules/L030.py

```diff
@@ -40,28 +39,10 @@ class Rule_L030(Rule_L010):
     """
 
     _target_elems: List[Tuple[str, str]] = [
-        ("type", "function_name"),
+        ("type", "function_name_identifier"),
         ("type", "bare_function"),
     ]
     _description_elem = "Function names"
 
     def _get_fix(self, segment, fixed_raw):
-        """Overrides the base class.
-
-        We need to do this because function_name nodes have a child
-        function_name_identifier that holds the actual name.
-        """
-        if segment.type == "function_name":
```

Comment:
I'm trying to remember -- I have a vague thought that maybe this was for BigQuery support, where there can be a project name, dataset name, AND function name. Does `git blame` show who did this?

If the above is correct, seems like we're missing a test case.

Reply:
It's possible the parse structure has changed since this code was added; maybe necessary at some point. I wouldn't stress too much about it. We keep refining, fixing, etc.

Reply:
I wouldn't like to name and shame anyone @barrywhart but have a look at above commit https://github.com/sqlfluff/sqlfluff/commit/93c4702804a63da6b39a7b77ecdcf4f515dbe74a :-)

Reply:
I think I named myself earlier. As much as I could without checking for myself. Another reason it is good to have a second Barry on the project! 🎉🤣

Reply:
Can never have enough Barrys on a project! 😁